packages/flutter_tools/lib/src/test/web_test_compiler.dart

```diff
@@ -118,24 +118,10 @@ class WebTestCompiler {
     required BuildInfo buildInfo,
     required WebRendererMode webRenderer,
   }) async {
-    LanguageVersion languageVersion = LanguageVersion(2, 8);
-    late final String platformDillName;
-
-    // TODO(zanderso): to support autodetect this would need to partition the source code into
-    // a sound and unsound set and perform separate compilations
-    final List<String> extraFrontEndOptions = List<String>.of(buildInfo.extraFrontEndOptions);
-    switch (buildInfo.nullSafetyMode) {
-      case NullSafetyMode.unsound || NullSafetyMode.autodetect:
-        // TODO(matanlurey): Should be unreachable, remove as part of https://github.com/flutter/flutter/issues/162846.
-        throw UnsupportedError('Unsound null safety mode is not supported');
-      case NullSafetyMode.sound:
-        languageVersion = currentLanguageVersion(_fileSystem, Cache.flutterRoot!);
-        platformDillName = 'ddc_outline.dill';
-    }
-
+    final LanguageVersion languageVersion = LanguageVersion(2, 8);
```

Comment:
The errors on CI are caused by this, 2.8 is the last non-null safe version. I _think_ if you just leave the language version string out, it will just work, but specifying an old version won't

Reply:
Right, it was my mistake, the original code was:

```dart
final LanguageVersion languageVersion = currentLanguageVersion(_fileSystem, Cache.flutterRoot!);
```

Is that better (or worse) than your suggestion of omission?